# Part D - Classification

# Reading and Pre-processing Data

In [16]:
import pandas as pd
dataframe=pd.read_csv('FW_Veg_Rem_Combined.csv')
df=dataframe.copy()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55367 entries, 0 to 55366
Data columns (total 43 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        55367 non-null  int64  
 1   Unnamed: 0.1      55367 non-null  int64  
 2   fire_name         25913 non-null  object 
 3   fire_size         55367 non-null  float64
 4   fire_size_class   55367 non-null  object 
 5   stat_cause_descr  55367 non-null  object 
 6   latitude          55367 non-null  float64
 7   longitude         55367 non-null  float64
 8   state             55367 non-null  object 
 9   disc_clean_date   55367 non-null  object 
 10  cont_clean_date   27477 non-null  object 
 11  discovery_month   55367 non-null  object 
 12  disc_date_final   28708 non-null  object 
 13  cont_date_final   25632 non-null  object 
 14  putout_time       27477 non-null  object 
 15  disc_date_pre     55367 non-null  object 
 16  disc_pre_year     55367 non-null  int64 

In [17]:
import numpy as np
df.dropna(axis='rows', how='any', inplace=True)
df.head()

Unnamed: 0  Unnamed: 0.1         fire_name  fire_size fire_size_class  \
3            3             3            WNA  1       1.00               B   
9            9             9             CRAZY       1.00               B   
26          26            26        STO3573155       2.00               B   
31          31            31       AXE RONDEAU       1.20               B   
35          35            35  FY2001-EARLY-021      30.18               C   

   stat_cause_descr  latitude  longitude state disc_clean_date  ... Wind_cont  \
3    Debris Burning   39.6414  -119.3083    NV        6/6/2005  ...       0.0   
9     Miscellaneous   48.8394   -99.7185    ND       4/19/2004  ...       0.0   
26   Debris Burning   35.7189   -92.3350    AR       1/19/2014  ...       0.0   
31   Debris Burning   48.8330   -99.7836    ND        4/7/2007  ...       0.0   
35   Debris Burning   31.2590   -84.8956    GA      10/19/2000  ...       0.0   

   Hum_pre_30 Hum_pre_15  Hum_pre_7 Hum_cont Prec_pre_30  Prec_pre_15  \
3   44.778429  37.140811  35.353846      0.0        10.4          7.2   
9   64.606509  55.943038  54.337838      0.0        12.3          1.8   
26  76.091398   0.000000   0.000000      0.0         0.0          0.0   
31  77.575012  75.963981  71.173116      0.0        21.2         21.2   
35  67.551783  61.733788  60.328571      0.0         0.0          0.0   

   Prec_pre_7 Prec_cont  remoteness  
3         0.0       0.0    0.487447  
9         0.0       0.0    0.291683  
26        0.0       0.0    0.227270  
31        9.2       0.0    0.292303  
35        0.0       0.0    0.162416  

[5 rows x 43 columns]

In [3]:
from sklearn.model_selection import train_test_split
train_set, test_set = train_test_split(df, test_size=0.30, random_state=568)
print(train_set.info())
print(test_set.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12341 entries, 30109 to 32049
Data columns (total 43 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        12341 non-null  int64  
 1   Unnamed: 0.1      12341 non-null  int64  
 2   fire_name         12341 non-null  object 
 3   fire_size         12341 non-null  float64
 4   fire_size_class   12341 non-null  object 
 5   stat_cause_descr  12341 non-null  object 
 6   latitude          12341 non-null  float64
 7   longitude         12341 non-null  float64
 8   state             12341 non-null  object 
 9   disc_clean_date   12341 non-null  object 
 10  cont_clean_date   12341 non-null  object 
 11  discovery_month   12341 non-null  object 
 12  disc_date_final   12341 non-null  object 
 13  cont_date_final   12341 non-null  object 
 14  putout_time       12341 non-null  object 
 15  disc_date_pre     12341 non-null  object 
 16  disc_pre_year     12341 non-null  in

# Features
I have choosen the following features as I want to predict the magnitude of the fire based on weather conditions.

X = "Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"

Y = "fire_size_class"

# Reason

*When I was using "fire_size" for the Y feature, I had gotten continuous value error. Hence, I changed to "fire_size_class" as suggested to using discreate values by creating buckets; which my database already has.*

# Decision Tree

## Training Set

In [26]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

X = train_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = train_set["fire_size_class"]

tree_classifier = DecisionTreeClassifier().fit(X,Y)
print(tree_classifier)

y_pred = tree_classifier.predict(X)
c_matrix = confusion_matrix(Y, y_pred)
print('The confusion Matrix is: ')
print(c_matrix)

DecisionTreeClassifier()
The confusion Matrix is: 
[[6106    1    0    0    1    5]
 [ 674 1389    1    0    1    4]
 [ 152    2  309    0    0    1]
 [  83    2    0  173    0    2]
 [ 296    6    2    3  737   22]
 [ 645   12    6    3   40 1663]]


In [25]:
from sklearn.metrics import accuracy_score, precision_score,recall_score, f1_score
print("The Evalution metrics are as follows: ")
print("Accuracy: ", accuracy_score(Y, y_pred))
print("Precision: ", precision_score(Y, y_pred, average="weighted"))
print("Sensitivity: ", recall_score(Y, y_pred, average="weighted"))
print("F1 Score: ", f1_score(Y, y_pred, average="weighted"))

The Evalution metrics are as follows: 
Accuracy:  0.8546038948761581
Precision:  0.8830948379567971
Sensitivity:  0.8546038948761581
F1 Score:  0.8504088999629371


# Result for Decision Tree

The Evalution metrics are as follows: 
1. Accuracy:  0.8408556843043513
1. Precision:  0.8718146288687685
1. Sensitivity:  0.8408556843043513
1. F1 Score:  0.8361067547616357

# Gaussian Naive Bayes

In [63]:
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, precision_score,recall_score, f1_score, confusion_matrix
from sklearn.metrics import confusion_matrix

# Training Set

X = train_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = train_set["fire_size_class"]

gnb = GaussianNB()
print(gnb.fit(X,Y))


GaussianNB()


In [51]:
y_pred = gnb.predict(X)
c_matrix = confusion_matrix(Y, y_pred)
print("Confusion Metrix: ")
print(c_matrix)

print("The Evalution metrics are as follows: ")
print("Accuracy: ", accuracy_score(Y, y_pred))
print("Precision: ", precision_score(Y, y_pred, average="weighted"))
print("Sensitivity: ", recall_score(Y, y_pred, average="weighted"))
print("F1 Score: ", f1_score(Y, y_pred, average="weighted"))



Confusion Metrix: 
[[5095    0    1    1   60  956]
 [1693    0    0    0   23  353]
 [ 360    0    1    0    0  103]
 [ 192    0    0    0    3   65]
 [ 668    0    0    0    2  396]
 [1431    0    0    0    8  930]]
The Evalution metrics are as follows: 
Accuracy:  0.48845312373389516
Precision:  0.3516650589256935
Sensitivity:  0.48845312373389516
F1 Score:  0.39405142725770587


C:\Users\s538103\Anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1221: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


# Random Forest Classifier

In [52]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, precision_score,recall_score, f1_score, confusion_matrix
from sklearn.metrics import confusion_matrix


# Training Set


In [55]:
X = train_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = train_set["fire_size_class"]
rfc = RandomForestClassifier()
print(rfc.fit(X,Y))

y_pred = rfc.predict(X)
c_matrix = confusion_matrix(Y, y_pred)
print("Confusion Metrix: ")
print(c_matrix)

print("The Evalution metrics are as follows: ")
print("Accuracy: ", accuracy_score(Y, y_pred))
print("Precision: ", precision_score(Y, y_pred, average="weighted"))
print("Sensitivity: ", recall_score(Y, y_pred, average="weighted"))
print("F1 Score: ", f1_score(Y, y_pred, average="weighted"))


RandomForestClassifier()
Confusion Metrix: 
[[6080   12    1    0    3   17]
 [ 663 1390    2    1    3   10]
 [ 151    1  306    0    1    5]
 [  82    0    0  170    2    6]
 [ 292    3    1    1  716   53]
 [ 632    7    2    1   12 1715]]
The Evalution metrics are as follows: 
Accuracy:  0.8408556843043513
Precision:  0.8699324335001136
Sensitivity:  0.8408556843043513
F1 Score:  0.8361995357658407


# Results



In [57]:

from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, precision_score,recall_score, f1_score, confusion_matrix
from sklearn.metrics import confusion_matrix

# Training Set
X = test_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
Y = test_set["fire_size_class"]

rfc = RandomForestClassifier()
print(rfc.fit(X,Y))

y_pred = rfc.predict(X)
c_matrix = confusion_matrix(Y, y_pred)
print("Confusion Metrix: ")
print(c_matrix)

print("The Evalution metrics are as follows: ")
print("Accuracy: ", accuracy_score(Y, y_pred))
print("Precision: ", precision_score(Y, y_pred, average="weighted"))
print("Sensitivity: ", recall_score(Y, y_pred, average="weighted"))
print("F1 Score: ", f1_score(Y, y_pred, average="weighted"))


RandomForestClassifier()
Confusion Metrix: 
[[2694    2    0    0    1    1]
 [ 296  620    0    0    0    1]
 [  54    0  106    0    0    0]
 [  42    1    0   87    0    1]
 [ 121    0    0    1  305   12]
 [ 230    1    0    0    5  708]]
The Evalution metrics are as follows: 
Accuracy:  0.8546038948761581
Precision:  0.8830276745796406
Sensitivity:  0.8546038948761581
F1 Score:  0.8504237672224091


In [59]:

# SVM

# import numpy as np
# from sklearn.svm import SVC

# # Training Set
# X = train_set[["Temp_pre_7","Wind_pre_7","Hum_pre_7","Prec_pre_7"]]
# Y = train_set["fire_size_class"]

# svm_classifier = SVC(kernel = 'linear')
# print(svm_classifier.fit(X,Y))

# Conclusion



# Results

**I have used Random Forest Classifier on the test set:**

The Evalution metrics are as follows: 
Accuracy:  0.8546038948761581
Precision:  0.8830948379567971
Sensitivity:  0.8546038948761581
F1 Score:  0.8504088999629371

*The values of the matrics can be better and increase the accuracy and other values to 90%+.*
